api/src/main/java/me/lokka30/treasury/api/economy/currency/Currency.java

```diff
@@ -1,236 +1,141 @@
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    @NotNull
-    public String @NotNull [] getCurrencyNames() {
-        return names;
-    }
+    boolean isDefault();
 
     /**
-     * Some economy providers like to round balances' decimals.
-     * Economy providers that do not round any digits should specify `-1`.
+     * Checks if this currency is the default currency in a specific world.
      *
-     * @return how many rounded digits the provider uses, or `-1` for none
-     * @author lokka30, MrIvanPlays
+     * @param world The name of the world to use for the check.
+     * @return True if this currency is the default currency for the specified world. This
+     * method should default to a global context if multi-world support is not present. If the world
+     * does not exist, this should default to the default world.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    public int getRoundedDigits() {
-        return roundedDigits;
-    }
+    boolean isDefault(@NotNull String world);
 
     /**
-     * Get the conversion coefficient of this currency.
+     * Used to convert this {@link Currency} to another based on a specified amount of the other
+     * currency.
      *
-     * @return conversion coefficient
-     * @author MrIvanPlays
+     * @param currency The currency we are converting to.
+     * @param amount The amount to be converted to the specified {@link Currency}
+     * @param subscription The {@link EconomySubscriber} accepting the resulting {@link Double} that
+     *                     represents the converted amount of the specified {@link Currency}.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    public double getConversionCoefficient() {
-        return conversionCoefficient;
-    }
+    void to(@NotNull Currency currency, double amount, @NotNull EconomySubscriber<Double> subscription);
 
     /**
-     * Sets a new conversion coefficient of this currency.
+     * Used to get the double representation of an amount represented by a formatted string.
      *
-     * @param conversionCoefficient new conversion coefficient
-     * @author MrIvanPlays
+     * @param formatted The formatted string to be converted to double form.
+     * @param subscription The {@link EconomySubscriber} accepting the resulting {@link Double} that
+     *                     represents the deformatted amount of the formatted String.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    protected void setConversionCoefficient(double conversionCoefficient) {
-        this.conversionCoefficient = conversionCoefficient;
-    }
+    void deformat(@NotNull String formatted, @NotNull EconomySubscriber<Double> subscription);
```

Comment:
I believe `parse` could be a better method name. Uncertain on that

@MrIvanPlays what do you think about this change?

Reply:
I believe currencies as of right now just need 1-2 small modifications to fulfill OP's needs rather an rewrite.